lib/util/vulnerability/utils.ts

```diff
@@ -0,0 +1,30 @@
+export function setHighestVulnerabilitySeverity<
+  T extends Record<string, any>,
+  TChild extends Record<string, any> | undefined
+>(config: Record<string, any>, parent: T, child: TChild): void {
```

Comment:
nit I would make this side-effect free. rename it `getHighestVulnerabilitySeverity` and allow the caller to set `config.vulnerabilitySeverity`.

Reply:
Good point - incorporated